### PR TITLE
Append \everypar toks to \g__para_standard_everypar_tl

### PR DESCRIPTION
**Pull request by muzimuzhi**
Fix rollback 2023/06/01 of \g__para_standard_everypar_tl, labeled "minipage fix". See #1386.

# Internal housekeeping

## Status of pull request

- Feedback wanted 
- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [ ] Rollback provided (if necessary)?
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: base/ltpara.dtx [hunk 1]
```diff
@@ -819,6 +819,18 @@
   \@@_handle_indent:
 % \the \everypar           % <--- done differently below
 }
+%    \end{macrocode}
+% \changes{v1.0n}{2024/06/21}
+%         {Append \cs{everypar} toks to \cs{g_@@_standard_everypar_tl},
+%          rollback 2023/06/01 (gh/1386)}
+%    \begin{macrocode}
+%<latexrelease>\cs_set:Npn \@@_tmp:w #1#2#3#4#5 { }
+%<latexrelease>\tl_gput_right:Nx \g_@@_standard_everypar_tl {
```
Comment (josephwright) [inline base/ltpara.dtx:828]: `\tl_gpur_right:Ne`

Comment (muzimuzhi) [reply base/ltpara.dtx:828]: Do you also suggest to change other two `\tl_gput_right:Nx`s in `ltpara.dtx`?

Comment (davidcarlisle) [reply base/ltpara.dtx]: @josephwright  I think it's Ok to use `e` (as @muzimuzhi  has already changed it) but this is rollback so rolling back to what it was would have been more natural  I think?

Comment (muzimuzhi) [reply base/ltpara.dtx]: Ah this is rollback.

I just checked that `\tl_gput_right:Ne` was not provided until l3kernel 2023-10-10 (see latex3/latex3@fe0eaaa7fbfc59ce9226c7613036a76ef9694c94). So maybe I should keep using `x` in that rollback code for 2023/06/01.

Comment (FrankMittelbach) [reply base/ltpara.dtx]: For rollback I suggest we stick to roll back to what we had back then and not do some thing like x to e (or whatever) even if it appears to be safe. Any unnecessary change would at least alter a test with `\CheckCommand` (even if that is unlikely). Of course they may be a few cases where we want to divert from that general rule, but as a best practice I think that is better.

Comment (davidcarlisle) [reply base/ltpara.dtx]: @muzimuzhi  sorry can you revert back to `x`

Comment (muzimuzhi) [reply base/ltpara.dtx:828]: Done. I dropped the 'x to e' commit https://github.com/latex3/latex2e/commit/1c23fde6c347c8d9f0e59657642e2256962819cd.

**Comment by muzimuzhi** [comment]
Need a ltnews40 entry?

File: base/changes.txt [hunk 2]
```diff
@@ -6,12 +6,19 @@ to completeness or accuracy and it contains some references to files that are
...
+	Append \everypar toks to \g__parar_standard_everypar_tl, rollback
+	2023/06/01 (gh/1386)
+
 2024-06-19  David Carlisle  <David.Carlisle@latex-project.org>
 
 	* fontdef.dtx:
 	preload ts1lmr.fd in Unicode engines
 
-2024-06-19 Joseph Wright  <Joseph.Wright@latex-project.org>
+2024-06-19  Joseph Wright  <Joseph.Wright@latex-project.org>
```
Comment (josephwright) [inline base/changes.txt:20]: Why the unrelated change here?

**Review by josephwright — APPROVED** [review]

File: base/changes.txt (hunk 2 above)
Comment (muzimuzhi) [reply base/changes.txt:20]: To unify whitespaces. I can drop this unrelated change.

File: base/changes.txt [hunk 3]
```diff
@@ -17,8 +17,7 @@ not part of the distribution.
 	* fontdef.dtx:
 	preload ts1lmr.fd in Unicode engines
 
-2024-06-19  Joseph Wright  <Joseph.Wright@latex-project.org>
-
```
Comment (FrankMittelbach) [inline base/changes.txt]: it has no been reverted, which isn't a problem either way, but I agree with @muzimuzhi 's approch @josephwright to clean up such things on the fly rather than making unnecessary pull requests for such a minor cleanup. I usually do that too when I notice one of those.

Comment (FrankMittelbach) [reply base/changes.txt]: can't type as usual. I meant: "If it has been reverted, so be it, but I agree with ...

**Comment by muzimuzhi** [comment]
The base branch of this PR is changed from `develop` to `main`, and I've moved ltnews entry to `ltnews39.tex`, adapted changes entry, and updated date.

File: base/doc/ltnews39.tex [hunk 4]
```diff
@@ -725,6 +725,19 @@ \subsection{Struts at the end of footnotes or \texttt{p} columns}
 \pkg{bigfoot})}
 
 
+\subsection{Fix a ``missing \cs{item}'' rollback error}
+
+If \LaTeX{} is rolled back to a date not newer than 2023/06/01 and
+older than 2024/06/01, any list-based environment would raise an error
```
Comment (FrankMittelbach) [inline base/doc/ltnews39.tex:731]: not newer 2023 but older than 2024/06 ? I think you meant something like

rolled back to a date between ...

maybe?

Comment (davidcarlisle) [reply base/doc/ltnews39.tex:731]: between also was not is as it's used with would so

If \LaTeX{} was rolled back to a date between  2023/06/01 and
2024/06/01, any list-based environment would raise an error

Comment (muzimuzhi) [reply base/doc/ltnews39.tex:731]: The original "not newer than 2023/06/01" should be "not older than ...".

To express the constraint "2023/06/01 <= date < 2024/06/01", how about

```suggestion
If \LaTeX{} is rolled back to a date between 2023/06/01 (inclusive) and
2024/06/01 (exclusive), any list-based environment would raise an error
```

Comment (FrankMittelbach) [reply base/doc/ltnews39.tex:731]: looks fine

**Review by FrankMittelbach — APPROVED** [review]

File: base/doc/ltnews39.tex [hunk 5]
```diff
@@ -725,6 +725,19 @@ \subsection{Struts at the end of footnotes or \texttt{p} columns}
 \pkg{bigfoot})}
 
 
+\subsection{Fix a ``missing \cs{item}'' rollback error}
+
+If \LaTeX{} is rolled back to a date between 2023/06/01 (inclusive) and
+2024/06/01 (exclusive), any list-based environment would raise an error
```
Comment (muzimuzhi) [inline base/doc/ltnews39.tex:731]: Maybe nowadays YYYY-MM-DD form is more preferable?

Comment (FrankMittelbach) [reply base/doc/ltnews39.tex:731]: hmm, I personally prefer the old one, but I don't mind either way

Comment (muzimuzhi) [reply base/doc/ltnews39.tex:731]: Then let it be.
